### ISSUE TITLE
Figure out how to handle severity label in alerting Prometheus Rule

### ISSUE THREAD

**Issue by fourstepper** [oskoperator/osko#103]
For example, in automatically generated issues, we already create two different severities in one alerting PrometheusRule

How to decide on what those should be? Opsgenie expects P1 - P4, PagerDuty might expect something else

**Comment by fourstepper** [comment]
That's an interesting idea, although I'm a bit worried about keeping OSKO up to date with all available providers and their severity levels.

To make it more clear for an outside perspective - we are trying to implement the multi-window, multi-burn-rate alerting here in an automatic fashion based on an existing SLO. In theory (and in our practice) having this as a config option would be good enough, but other usecases might differ.

**Comment by Hy3n4** [comment]
I would use the defaults as `critical`, `high-fast`, `high-slow`, `low`, and `no_slo`. This is taken from the SRE book from the alerting chapter. They are using it to sort the SLOs, but we can also use it as a default severities.

**Comment by Hy3n4** [comment]
Do we think we'll ever need to use more than one alerting tool at a time? 🤔

**Comment by fourstepper** [comment]
The defaults are one thing, but how to configure these is a whole other thing that I think is the bigger issue here 🤔

Regarding your question, that's really hard to say. I would focus on supporting one at a time for now. If we can stay backwards compatible with supporting one, setting the various severities in the config, for example, we can always add an option of multiple ones down the line.

**Comment by Hy3n4** [comment]
something like this?

```
type AlertSeverities struct {
	Critical string
	HighFast string
	HighSlow string
	Low      string
	NoSlo    string
}

alertingTools := []AlertingTool{
    {
	    Name: "opsgenie",
	    SeverityMap: map[string]string{
		    "critical":  "P1",
		    "high-fast": "P2",
		    "high-slow": "P3",
		    "low":       "P4",
		    "no-slo":    "P5",
	    },
    },
    {
	    Name: "pagerduty",
	    SeverityMap: map[string]string{
		    "critical":  "SEV-1",
		    "high-fast": "SEV-2",
		    "high-slow": "SEV-3",
		    "low":       "SEV-4",
		    "no-slo":    "SEV-5",
	    },
    },
}
```

**Comment by fourstepper** [comment]
Eh, I mean I guess we could do that :) Alternatively, we could just set this in the `internal/config/config.go`

**Comment by Hy3n4** [comment]
Yes, I copy-pasted it from that file :D

**Comment by fourstepper** [comment]
Yeah, makes sense. Just not sure if we want to map these out directly.

Alternatively, we could have some "custom" type for those not using either of the ones we support (like opsgenie and pagerduty from the above example)

**Comment by Hy3n4** [comment]
we can allow configuring it through the env vars, but it would not cover the cases where there is a need for multiple alerting tools for one triggered alert.

**Comment by fourstepper** [comment]
I wouldn't cover that for now, tbh

**Comment by Hy3n4** [comment]
Working through PagerDuty support this week turned up a concrete answer to this, plus one thing the question didn't anticipate.

## The vocabulary is already selectable

`osko.dev/alertingTool` on an SLO, falling back to `OSKO_ALERTING_TOOL`, picks the severity vocabulary — read at `internal/helpers/prometheus_helper.go` and resolved in `internal/config/utils.go`:

| tool | PageCritical / PageHigh / TicketHigh / TicketMedium |
| --- | --- |
| `opsgenie` (default) | `P1` / `P2` / `P3` / `P4` |
| `pagerduty` | `SEV_1` / `SEV_2` / `SEV_3` / `SEV_4` |
| `custom` | `OSKO_ALERTING_SEVERITY_CRITICAL` / `_HIGH` / `_MEDIUM` / `_LOW` |

This has worked for a while but was undocumented. #159 adds it to `docs/labels-and-annotations.md`.

## The thing the question didn't anticipate

**`SEV_1`–`SEV_4` are not valid PagerDuty severities.** PagerDuty's Events API v2 accepts only `critical`, `error`, `warning`, `info` for `payload.severity` — the PD-CEF enum. So this, which is the obvious wiring, is rejected by PagerDuty:

```yaml
pagerduty_configs:
  - severity: '{{ .CommonLabels.severity }}'   # sends "SEV_1" → rejected
```

So the answer to "how do we decide what the severity should be?" is that **OSKO should not try to decide the receiver's severity at all.** The label is a *routing key*; the receiver's own severity is a property of the receiver and belongs in the Alertmanager config:

```yaml
route:
  routes:
    - matchers: ['severity="P1"']
      receiver: pd-p1
    - matchers: ['severity="P2"']
      receiver: pd-p2

receivers:
  - name: pd-p1
    pagerduty_configs:
      - routing_key: '<key>'
        severity: critical      # PD's vocabulary, not OSKO's
  - name: pd-p2
    pagerduty_configs:
      - routing_key: '<key>'
        severity: error
```

That pattern is validated against a live Mimir 3.2.1 Alertmanager (accepted, read back, anchors resolved) and is written up in `docs/alertmanager-configs.md` in #159.

Same shape works for ntfy via `Priority` headers, and for Opsgenie via its own priority field. The `P1`–`P4` labels never need to match any vendor's vocabulary.

## A bug found while documenting this

`internal/config/utils.go` had `PageHigh` and `TicketHigh` both reading `OSKO_ALERTING_SEVERITY_HIGH`. Different defaults masked it, so it only bit once that variable was set — at which point P2 and P3 collapse to the same value and Alertmanager can neither route nor inhibit between them. Fixed in #159, with a test asserting no two tiers ever resolve to the same value.

## What's left to decide

The `pagerduty` → `SEV_*` mapping is misleading: it names a vendor whose API rejects those exact values, which invites the broken `severity: '{{ .CommonLabels.severity }}'` wiring above. Options:

1. Drop the `pagerduty` entry — PagerDuty users select `opsgenie`'s `P1`–`P4` or `custom`, and set the real severity per receiver.
2. Rename the values to something obviously routing-only (`sev-1`, or reuse `P1`–`P4`).
3. Keep it and rely on documentation.

Changing it is breaking for anyone routing on `SEV_*` today, so it wants a deliberate call rather than a drive-by. Leaving this open for that decision — the "how do we decide" question itself is answered above.

**Comment by fourstepper** [comment]
@Hy3n4 not sure how this moves us forward. I guess that the PagerDuty's map is different now?
